### PR TITLE
Add detection for multiple emails for gitlab password reset

### PR DESCRIPTION
**Pull request by melenevskyi**
### Background

Attackers are exploiting a Critical (CVSS 10.0) GitLab vulnerability in which user account password reset emails could be delivered to an unverified email address.

### CODE REVIEW

File: rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        return False
+
+    if len(emails) > 1:
+        return True
+    return False
+
+
+def title(event):
+    emails = event.deep_get("detail", "target_details", default="")
+    return f"Someone tried to reset your password with multiple emails :{emails}"
```
Comment (egibs) [inline rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.py:23]: Can this be updated to be something like:
```py
return f"[GitLab] Multiple password reset emails requested for {emails}"
```?

Comment (melenevskyi) [reply rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.py:23]: sure!

**Review by egibs — APPROVED** [review]
LGTM!

File: rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+AnalysisType: rule
+Filename: gitlab_audit_password_reset_multiple_emails.py
+RuleID: "GitLab.Audit.Password.Reset.Multiple.Emails"
+DisplayName: "GitLab Audit Password Reset Multiple Emails"
```
Comment (arielkr256) [inline rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.yml:4]:
```suggestion
DisplayName: "CVE-2023-7028 - GitLab Audit Password Reset Multiple Emails"
```

File: rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+Enabled: True
+LogTypes:
+  - GitLab.Audit
+Tags:
+  - GitLab
```
Comment (arielkr256) [inline rules/gitlab_rules/gitlab_audit_password_reset_multiple_emails.yml:9]:
```suggestion
  - GitLab
  - CVE-2023-7028
```

File: rules/gitlab_rules/gitlab_production_password_reset_multiple_emails.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+AnalysisType: rule
+Filename: gitlab_production_password_reset_multiple_emails.py
+RuleID: "GitLab.Production.Password.Reset.Multiple.Emails"
+DisplayName: "GitLab Production Password Reset Multiple Emails"
```
Comment (arielkr256) [inline rules/gitlab_rules/gitlab_production_password_reset_multiple_emails.yml:4]:
```suggestion
DisplayName: "CVE-2023-7028 - GitLab Production Password Reset Multiple Emails"
```

File: rules/gitlab_rules/gitlab_production_password_reset_multiple_emails.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Enabled: True
+LogTypes:
+  - GitLab.Production
+Tags:
+  - GitLab
```
Comment (arielkr256) [inline rules/gitlab_rules/gitlab_production_password_reset_multiple_emails.yml:9]:
```suggestion
  - GitLab
  - CVE-2023-7028
```

**Pull request merged**